torch_xla/core/xla_model.py

```diff
@@ -109,10 +114,15 @@ def get_ordinal(defval=0):
   Returns:
     The replication ordinal of the current thread.
   """
-  if pjrt.using_pjrt():
-    return pjrt.global_ordinal()
+  global g_ordinal
```

Comment:
I think this will break PJRT + v3 cases, the implementation we had checks the devices in
```
  m.def("_xla_get_default_device_ordinal", []() {
    std::string device_str = GetCurrentThreadDevice();
    torch::lazy::BackendDevice device =
        bridge::AtenDeviceToXlaDevice(device_str);
    return device.ordinal();
  });
```

Reply:
Hmm, this is confusing. That call is in the C++ layer. Then allow_in_graph won't work here.

But we can work around by caching a map...

Reply:
I am not sure actually, effectively this function won't return constant in the v3 cases because there are two devices per process. This is a bit tricky.

Reply:
I wonder if we can bypass the v3 cases for now, what's going to happen if you add a condition here to skip this cahce value of we are on v3 + PJRT?

Reply:
It will introduce graph breaks in Dynamo.

Reply:
Should this use thread local storage instead?

Reply:
That's cool. Was not aware python has this feature. Let me work on it.

Reply:
Dynamo doesn't seem to compile in the same thread as the user code. threading.local doesn't work here.